File: 프로그래머스/lv0/120844. 배열 회전시키기/배열 회전시키기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> numbers, string direction)
{
    vector<int> answer;
    if (direction == "left")
    {
        int temp = numbers[0];
        for (int i = 1; i < numbers.size(); i++)
            answer.push_back(numbers[i]);
        answer.push_back(temp);
    }
    if (direction == "right")
    {
        int temp = numbers[numbers.size()-1];
        for (int i = numbers.size()-2; i >= 0; i--)
            answer.insert(answer.begin(), numbers[i]);
        answer.insert(answer.begin(), temp);
        
    }

    return answer;
}

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.java
import java.util.*;

class Solution {
    public int[] solution(int[] numbers, String direction) {
        int[] answer = new int[numbers.length];
        
        Integer[] boxed = new Integer[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            boxed[i] = numbers[i];
        }
        Collections.rotate(Arrays.asList(boxed), direction.equals("right") ? 1 : -1);
        
        for (int i = 0; i < numbers.length; i++) {
            answer[i] = boxed[i];
        }
        
        return answer;
    }
}